projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php

```diff
@@ -366,3 +368,100 @@ function wpcom_edit_site_menu_override( $wp_admin_bar ) {
 	}
 }
 add_action( 'admin_bar_menu', 'wpcom_edit_site_menu_override', 41 );
+
+/**
+ * Adds site badges and plan information to the site title dropdown menu.
+ *
+ * @param WP_Admin_Bar $wp_admin_bar The WP_Admin_Bar core object.
+ */
+function wpcom_add_site_badges_and_plan( $wp_admin_bar ) {
+	// Get the current blog ID
+	$blog_id = get_current_blog_id();
+	$status  = new Status();
+
+	// Check for various site types
+	$badge_text = '';
+
+	// Check if this is a P2 site
+	if ( str_contains( get_stylesheet(), 'pub/p2' ) ||
+		( function_exists( '\WPForTeams\is_wpforteams_site' ) &&
+		\WPForTeams\is_wpforteams_site( $blog_id ) ) ) {
+		$badge_text = 'P2';
+	} elseif ( (bool) get_option( 'wpcom_is_staging_site' ) ) {
+		// Check for staging site
+		$badge_text = __( 'Staging', 'jetpack-mu-wpcom' );
+	} elseif ( function_exists( 'wpcom_site_has_feature' ) && wpcom_site_has_feature( 'trial' ) ) {
+		// Check for trial site
+		$badge_text = __( 'Trial', 'jetpack-mu-wpcom' );
+	} elseif ( ( get_option( 'launch-status' ) !== 'launched' ) || $status->is_coming_soon() ) {
+		// Check for Coming Soon site
+		$badge_text = __( 'Coming Soon', 'jetpack-mu-wpcom' );
+	} elseif ( $status->is_private_site() ) {
+		// Check for private site
+		$badge_text = __( 'Private', 'jetpack-mu-wpcom' );
+	} elseif ( ( function_exists( 'has_blog_sticker' ) && has_blog_sticker( 'difm-lite-in-progress' ) ) ||
+		( function_exists( 'wpcomsh_is_site_sticker_active' ) && wpcomsh_is_site_sticker_active( 'difm-lite-in-progress' ) ) ) {
+		// Check for Express service
+		$badge_text = __( 'Express', 'jetpack-mu-wpcom' );
+	} elseif ( function_exists( 'is_simple_site_redirect' ) && is_simple_site_redirect( $status->get_site_suffix() ) ) {
+		// Check for Redirect site
+		$badge_text = __( 'Redirect', 'jetpack-mu-wpcom' );
+	} elseif ( ! empty( get_option( 'options' )['is_domain_only'] ) ) {
+		// Check for Domain Only site
+		$badge_text = __( 'Domain Only', 'jetpack-mu-wpcom' );
+	}
+
+	// Add badge to the site name dropdown if a badge is applicable
+	$status_text = '';
+	if ( $badge_text ) {
+		$status_text = '<div class="wp-admin-bar__site-info">
+							<span class="wp-admin-bar__site-info-label">' . __( 'Status', 'jetpack-mu-wpcom' ) . '</span>
+							<div class="wp-admin-bar__info-badges">' . esc_html( $badge_text ) . '</div>
+						</div>';
+	}
+
+	// Add plan information for non-staging sites
+	$plan_text  = '';
+	$is_staging = (bool) get_option( 'wpcom_is_staging_site' );
+	if ( ! $is_staging ) {
+		$current_plan = Current_Plan::get();
```

Comment:
I'm not sure but it seems better to use the `WPCOM_Store_API::get_current_plan` to get the current plan because the `Current_Plan::get` doesn't seem to work on simple sites 🤔

```php
require_once WP_CONTENT_DIR . '/admin-plugins/wpcom-billing.php';
$current_plan = WPCOM_Store_API::get_current_plan( get_current_blog_id() );
```

Reply:
This fixes the issue Thank you. I wonder how we can avoid "guessing" the right things for this kind of things.

Reply:
I always learned from the codebase 🙈 But things related to WP.com might be different between simple and atomic sites.